plugins/in_tail/tail_file.c

```diff
@@ -1560,6 +1570,22 @@ char *flb_tail_file_name(struct flb_tail_file *file)
     if (strstr(buf, "\\\\?\\")) {
         memmove(buf, buf + 4, len + 1);
     }
+#elif defined(FLB_SYSTEM_FREEBSD)
+    if ((file_entries = kinfo_getfile(getpid(), &count)) == NULL) {
+        flb_free(buf);
+        return NULL;
+    }
+
+    for (idx=0; idx < count; idx++) {
+        info = &file_entries[idx];
+        if (info->kf_fd == file->fd) {
+            len = strlen(info->kf_path);
```

Comment:
Could you please use `strncpy` here instead? Just ensure that you use `sizeof(buf) - 1` as `len` and truncate the buffer afterwards just in case `buf[sizeof(buf) - 1] = 0;` 

Additionally, if you could use more descriptive names for the variables you are adding (eg. `file_index` instead of `idx`, `file_count` instead of `count` and `file_entry` instead of `info` ) that would be awesome. 

Although you might be able to skip `info` altogether if you just access the entry by index since the line length and complexity is pretty low in that chunk of code.

On a side note, I noticed the apple specific code repeats the `strlen`, `memcpy` pattern even adding a stack allocated variable so I understand you used the same construct and I think we need to fix that piece of code to use buf with the appropriate size and nulling on error but that's for a different PR so thing aren't mixed up.

Reply:
I've made the requested changes.   Although "sizeof(buf)" is incorrect,  It needs to be PATH_MAX instead (buf is a pointer, not a character array).

Reply:
You're right about the sizeof thing, sorry, I mixed `buf` with `path` there.

Thanks a lot for making those changes, once you change the `flb_free` call this PR will be good to go for me.